cpp/src/arrow/dataset/scanner_test.cc

```diff
@@ -1254,8 +1261,32 @@ TEST(ScanNode, DeferredFilterOnPhysicalColumn) {
               ResultWith(UnorderedElementsAreArray(expected)));
 }
 
-TEST(ScanNode, ProjectionPushdown) {
-  // ensure non-projected columns are dropped
+TEST(ScanNode, DISABLED_ProjectionPushdown) {
```

Comment:
Disabled because of that TODO above?

Reply:
Correct, I'll add the JIRA number here once I file the appropriate follow up